RoslynPluginGenerator/HardcodedConstantSqaleGenerator.cs

```diff
@@ -0,0 +1,88 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="SqaleGenerator.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarLint.Common.Sqale;
+using SonarLint.XmlDescriptor;
+using SonarQube.Plugins.Common;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace SonarQube.Plugins.Roslyn
+{
+    /// <summary>
+    /// Generates a simple SQALE description using a constant value for each rule
```

Comment:
This class uses the data classes defined in the SonarLint assembly to build a sqale model. These classes are already XML-serializable, so the file is created just by using the XMLSerializer against the model (this is what the C# plugin does).